cranelift/filetests/filetests/isa/aarch64/wasm/load_store_dynamic_kind_i32_index_0_guard_no_spectre_i32_access_0x1000_offset.wat

```diff
@@ -0,0 +1,76 @@
+;;! target = "aarch64"
+;;!
+;;! settings = ['enable_heap_access_spectre_mitigation=false']
+;;!
+;;! compile = true
+;;!
+;;! [globals.vmctx]
+;;! type = "i64"
+;;! vmctx = true
+;;!
+;;! [globals.heap_base]
+;;! type = "i64"
+;;! load = { base = "vmctx", offset = 0, readonly = true }
+;;!
+;;! [globals.heap_bound]
+;;! type = "i64"
+;;! load = { base = "vmctx", offset = 8, readonly = true }
+;;!
+;;! [[heaps]]
+;;! base = "heap_base"
+;;! min_size = 0x10000
+;;! offset_guard_size = 0
+;;! index_type = "i32"
+;;! style = { kind = "dynamic", bound = "heap_bound" }
+
+;; !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+;; !!! GENERATED BY 'make-load-store-tests.sh' DO NOT EDIT !!!
+;; !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+(module
+  (memory i32 1)
+
+  (func (export "do_store") (param i32 i32)
+    local.get 0
+    local.get 1
+    i32.store offset=0x1000)
+
+  (func (export "do_load") (param i32) (result i32)
+    local.get 0
+    i32.load offset=0x1000))
+
+;; function u0:0:
+;; block0:
+;;   mov w9, w0
```

Comment:
Also, I'm going back and forth on whether I like having so many individual files or larger consolidated files. On the one hand, I understand why you did it this way (easier to write the shell script to generate them, and it's nice to have separate test cases in separate files for running them individually, and it's easier to write the parser to get the settings once from the top of the file). On the other hand, it might be more approachable overall to have a single large file of autogenerated test cases, rather than a massive directory with intimidating filenames. Thoughts?

Reply:
> One thought: could you say a bit more about what level of scrutiny all of the golden outputs have had?

I've spot checked a few of them, mostly Wasm-to-CLIF outputs, definitely haven't gone through all of them.

I think the primary value is for the future: they really shouldn't change unless we expect them to, and when we do improve these sequences we can give the ones that changed a deeper level of scrutiny.

Reply:
A suggestion: It might help to land as many of the wasm tests as possible separately, before this PR removes Cranelift's heap support, so we can see whether this PR changes the generated sequences. I think we can't merge any test changes involving the existing `heap_addr` CLIF instruction.

But filetests which go from .wat to native assembly should provide good information about the effect of this PR. That way the golden outputs initially are "whatever we do today", and their value as regression tests begins immediately.

Can filetests invoke specific legalization passes? The new wasm-to-CLIF filetests in this PR could also be added beforehand if we can test what the generated CLIF looks like after legalizing heap instructions.

Reply:
I don't think it can invoke specific legalization passes, but it can test the output of legalization as a whole if you input clif ir rather than wasm.

Reply:
I've pulled out the tests to https://github.com/bytecodealliance/wasmtime/pull/5439 so we can check for regressions and perturbations in this PR after I rebase.